### PR TITLE
CLOUDP-197016: Add atlas deployments search index create

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

[CLOUDP-197016](https://jira.mongodb.org/browse/CLOUDP-197016)

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

## Further comments

Flag flow
```shell

~/workplace/mongocli CLOUDP-197016 ⇡1 *26 !1 ?1 ❯ atlasdev deployments search index create Cluster43140 --type atlas --deploymentName Cluster07293 --collection sample_mflix --db embedded_movies                                                                  1.20 14:29:39
Search index created with ID: 650c454813937d0999cc36df
~/workplace/mongocli CLOUDP-197016 ⇡1 *26 !1 ?1 ❯ atlasdev deployments search index list  --deploymentName Cluster07293 --collection sample_mflix --db embedded_movies                                                                                             1.20 14:29:45
ID                         NAME           DATABASE          COLLECTION
650c454813937d0999cc36df   Cluster43140   embedded_movies   sample_mflix
```

### CODE REVIEW

File: internal/cli/atlas/deployments/search/indexes/create.go [hunk 1]
```diff
@@ -87,10 +112,37 @@ func (opts *CreateOpts) Run(ctx context.Context) error {
...
 		return err
 	}
-	return nil
+
+	opts.index, err = opts.store.CreateSearchIndexes(opts.ConfigProjectID(), opts.DeploymentName, index)
+	return err
+}
+
+func (opts *CreateOpts) Run(ctx context.Context) error {
+	if strings.EqualFold(opts.DeploymentType, "local") {
```
Comment (blva) [inline internal/cli/atlas/deployments/search/indexes/create.go:133]: TODO: Add question asking type here once start/stop PR is merged

File: internal/cli/atlas/deployments/search/indexes/create.go [hunk 2]
```diff
@@ -87,10 +112,43 @@ func (opts *CreateOpts) Run(ctx context.Context) error {
...
+	opts.index, err = opts.store.CreateSearchIndexes(opts.ConfigProjectID(), opts.DeploymentName, index)
+	return err
+}
+
+func (opts *CreateOpts) Run(ctx context.Context) error {
+	if opts.DeploymentType == "" {
+		if err := opts.PromptDeploymentType(); err != nil {
+			return err
+		}
+	}
```
Comment (andreaangiolillo) [inline internal/cli/atlas/deployments/search/indexes/create.go:137]: we should ask for deployment type first

File: internal/cli/atlas/deployments/search/indexes/create.go [hunk 3]
```diff
@@ -87,10 +112,43 @@ func (opts *CreateOpts) Run(ctx context.Context) error {
...
+
+	if strings.EqualFold(opts.DeploymentType, "local") {
```
Comment (andreaangiolillo) [inline internal/cli/atlas/deployments/search/indexes/create.go:139]:
```suggestion
	if strings.EqualFold(opts.DeploymentType, options.Local) {
```

File: internal/cli/atlas/deployments/search/indexes/create.go [hunk 4]
```diff
@@ -87,10 +112,43 @@ func (opts *CreateOpts) Run(ctx context.Context) error {
...
+		return opts.RunLocal(ctx)
+	}
+
+	if !opts.IsCliAuthenticated() {
+		return ErrNotAuthenticated
+	}
+
+	if opts.DeploymentName == "" {
+		return ErrNoDeploymentName
+	}
```
Comment (andreaangiolillo) [inline internal/cli/atlas/deployments/search/indexes/create.go:149]: is this error possible? if their did not provide the deployment name they will get the prompt right?

Comment (blva) [reply internal/cli/atlas/deployments/search/indexes/create.go:149]: ah, yes not anymore - thanks for catching

File: internal/cli/atlas/deployments/search/indexes/create.go (hunk 2 above)
Comment (blva) [reply internal/cli/atlas/deployments/search/indexes/create.go:137]: sorry, got confused here, isn't this the first thing the Run method is doing?

Comment (andreaangiolillo) [reply internal/cli/atlas/deployments/search/indexes/create.go:137]: ops, you are right 🤦 resolving xD

File: internal/cli/atlas/deployments/search/indexes/create.go (hunk 4 above)
Comment (blva) [reply internal/cli/atlas/deployments/search/indexes/create.go:149]: ah, no, nevermind, we should keep this, for Atlas right? as we don't list atlas clsters yet we ask for user to provide the name 

~/workplace/mongocli CLOUDP-197016 ⇡2 *27 +2 !1 ?2 ❯ atlasdev deployments search index create                                                                                                                                                                 35s  1.20 15:06:31
? What type of deployment would you like to work with? atlas
Error: deployment name is required for Atlas resources

File: internal/cli/atlas/deployments/search/indexes/create.go (hunk 3 above)
Comment (blva) [reply internal/cli/atlas/deployments/search/indexes/create.go:139]: will add in another PR

**Pull request merged**